### PR TITLE
Assert on total wait time in lock contention tests

### PR DESCRIPTION
**Pull request by krk**
### Description
Fixes LockTests disabled in https://github.com/async-profiler/async-profiler/pull/1006 by verifying a simpler metric, total lock wait time.

### Related issues
https://github.com/async-profiler/async-profiler/pull/982

### Motivation and context
Tests are fixed by not asserting on comparison to a constant value but comparing within the test results.

### How has this been tested?
`make test` in linux and mac.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: test/one/profiler/test/Assert.java [hunk 1]
```diff
@@ -5,17 +5,37 @@
 
 package one.profiler.test;
 
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 public class Assert {
+    private static final Logger log = Logger.getLogger(Runner.class.getName());
```
Comment (apangin) [inline test/one/profiler/test/Assert.java:12]: Why `Runner`? Perhaps `Assert.class`?

File: test/one/profiler/test/TestProcess.java [hunk 2]
```diff
@@ -85,6 +87,10 @@ public TestProcess(Test test, Os currentOs, String logDir) throws Exception {
         }
     }
 
+    public TestProcess dup() throws Exception {
+        return new TestProcess(this.test, this.currentOs, this.logDir);
+    }
```
Comment (apangin) [inline test/one/profiler/test/TestProcess.java:92]: Is it really used?

File: test/one/profiler/test/TestProcess.java [hunk 3]
```diff
@@ -66,6 +67,7 @@ public TestProcess(Test test, Os currentOs, String logDir) throws Exception {
         this.currentOs = currentOs;
         this.logDir = logDir;
         this.inputs = test.inputs();
+        this.test = test;
```
Comment (apangin) [inline test/one/profiler/test/TestProcess.java:70]: Minor stylistic comment: things being equal, arguments, fields and initialization should follow the same order.

File: test/test/lock/LockTests.java [hunk 4]
```diff
@@ -48,4 +19,16 @@ public void datagramSocketLock(TestProcess p) throws Exception {
...
         assert out.samples("sun/nio/ch/DatagramChannelImpl.send") > 10;
     }
+
+    @Test(mainClass = RaceToLock.class, inputs = { "0" }, args = "1000", output = true)
+    @Test(mainClass = RaceToLock.class, inputs = { "10000" }, args = "1000", output = true)
+    @Test(mainClass = RaceToLock.class, inputs = { "1000000000" }, args = "1000", output = true)
+    public void raceToLocks(TestProcess p) throws Exception {
+        int interval = Integer.parseInt(p.inputs()[0]);
+
+        p.profile("--lock " + interval + " --threads -o collapsed");
```
Comment (apangin) [inline test/test/lock/LockTests.java:29]: The test does not check profiling results at all, is it intentional?

File: test/test/lock/RaceToLock.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    private final Lock sharedLock = new ReentrantLock(true);
+    private final Object sharedObj = new Object();
+    private long sharedWaitTime = 0;
+
+    private final Random random = new Random();
+    private final Lock lock1 = new ReentrantLock(true);
+    private final Lock lock2 = new ReentrantLock(true);
+    private volatile long randomWaitTime = 0;
+
+    private volatile boolean exitRequested = false;
```
Comment (apangin) [inline test/test/lock/RaceToLock.java:26]: nit: it does not make difference here, but generally initializing volatile fields with default values is a bad practice.

File: test/test/lock/LockTests.java [hunk 6]
```diff
@@ -48,4 +19,16 @@ public void datagramSocketLock(TestProcess p) throws Exception {
         out = p.profile("-e lock -d 3 -o collapsed");
         assert out.samples("sun/nio/ch/DatagramChannelImpl.send") > 10;
     }
+
+    @Test(mainClass = RaceToLock.class, inputs = { "0" }, args = "1000", output = true)
+    @Test(mainClass = RaceToLock.class, inputs = { "10000" }, args = "1000", output = true)
+    @Test(mainClass = RaceToLock.class, inputs = { "1000000000" }, args = "1000", output = true)
```
Comment (apangin) [inline test/test/lock/LockTests.java:25]: Do we need to pass arguments, if they are always the same?

File: test/test/lock/LockTests.java (hunk 4 above)
Comment (krk) [reply test/test/lock/LockTests.java:29]: Yes, currently it is only checking the wait times in stdout of the test app.

File: test/one/profiler/test/TestProcess.java (hunk 2 above)
Comment (krk) [reply test/one/profiler/test/TestProcess.java:92]: not in the latest version, will remove.

File: test/test/lock/LockTests.java (hunk 6 above)
Comment (krk) [reply test/test/lock/LockTests.java:25]: Removing the arg, we can use 1s for all cases.

File: test/test/lock/RaceToLock.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.concurrent.locks.LockSupport;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class RaceToLock {
+
+    private final Lock sharedLock = new ReentrantLock(true);
+    private final Object sharedObj = new Object();
```
Comment (apangin) [inline test/test/lock/RaceToLock.java:18]: unused?

File: src/converter/Main.java [hunk 8]
```diff
@@ -113,6 +113,7 @@ private static void usage() {
                 "Flame Graph options:\n" +
                 "     --title STRING     Flame Graph title\n" +
                 "     --minwidth X       Skip frames smaller than X%\n" +
+                "     --grain X          Coarsen Flame Graph to the given grain size\n" +
```
Comment (apangin) [inline src/converter/Main.java:116]: Can you please rebase the PR on top of master, or merge master in your branch with a merge commit so that unrelated changes do not appear in diffs?

Comment (apangin) [reply src/converter/Main.java:116]: Never mind, it integrates correctly.
